nvflare/app_common/metrics_exchange/memory_metric_relayer.py

```diff
@@ -0,0 +1,73 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from queue import Queue
+
+from nvflare.apis.event_type import EventType
+from nvflare.apis.fl_context import FLContext
+from nvflare.app_common.metrics_exchange.metrics_exchanger import MetricsExchanger
+from nvflare.fuel.utils.constants import Mode
+from nvflare.fuel.utils.pipe.memory_pipe import MemoryPipe
+from nvflare.fuel.utils.pipe.pipe_handler import PipeHandler
+
+from .metric_relayer import MetricRelayer
+
+
+class MemoryMetricRelayer(MetricRelayer):
```

Comment:
keep old name "MetricsRetriever",

Add note it only exists for backward compatibility